### PR TITLE
Update libtiff to version 4.0.8

### PR DESCRIPTION
**Pull request by MaartenBent**
I had a go at updating libtiff. 

I used the following steps:
* Replaced the content of the tiff folder with the latest release (kept the two config files).
* Update `tif_config.h`. It is a copy of `tif_config.h.in` with the required defines filled in for Windows and OS X, 32 and 64 bit. Endianness (intel vs ppc) is also taken care of. Also note that most of the defines are not used in the tiff sources at all.
* `tiffconf.h` is obsolete, it has the same defines as `tif_config.h`. It is still included in the tiff sources, so let it include `tif_config.h`.
* Combined all the custom changes to the libtiff sources and committed them (skipped VC6-only changes).

And should all the `.\" $Id: ` lines be removed again? I suggest not, keeping the sources as original as possible makes updating easier.

For reference, the following will remove the lines automatically when running from the `src/tiff/libtiff` directory:
```bash
find . -type f \( -name "*.h*" -or -name "*.c*" \) | xargs sed -i '/^\/\* $Id:/ { N; d; }'
```

### CODE REVIEW

File: src/tiff/libtiff/tif_config.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#ifdef __APPLE__
+﻿#ifdef __APPLE__
```
Comment (MaartenBent) [inline src/tiff/libtiff/tif_config.h]: Seems I messed up the BOM here...

File: src/tiff/libtiff/tif_config.h [hunk 2]
```diff
@@ -381,26 +437,9 @@
 # endif
 #endif
 
-#if !defined (__BORLANDC__) && !defined (__WATCOMC__)
```
Comment (MaartenBent) [inline src/tiff/libtiff/tif_config.h]: Should this be re-added?

**Comment by MaartenBent** [comment]
I updated the config files. The `__APPLE__` part should be relevant for Xcode. Can someone using XCode verify it still works?

**Comment by vadz** [comment]
Thanks again for your work on this, it helped me with my changes, but now that they're done, I'm closing this in favour of #601.

We still really,really need to do something about the `tif_config.h`/`tiffconf.h` mess though. The cleanest would probably be to add separate `{tiffconf,tif_config}.xcode.h` headers and include either them or the `.vc.h` versions instead of including the normal ones.

The main problem here is that we need to distinguish between MinGW using configure (which should include normal headers) and MinGW using makefiles (which should include VC versions) and currently there is no way to do it AFAICS...

**Pull request closed without merging**